### PR TITLE
feat (JooqPaginate): Support multi condition query

### PR DESCRIPTION
**Pull request by mayangzz**
Support multi condition query And Optimize count implementation

### CODE REVIEW

**Comment by mayangzz** [comment]
Features :
1. At this time, the paging query In contrast to Jooq's native support for the functionality, the Next Jooq plugin currently does not allow multi-conditional order by sorting; this pr adds an overloaded method to support.

2. The paging count query has no restrictions or limitations; however, if the query is in a large table, the query will be quite slow.
In order to determine how many rows overall this query returned. This pr offered a count method, but doing so necessitates a separate count query.

File: plugins/cloudopt-next-jooq/src/main/kotlin/net/cloudopt/next/jooq/JooqPaginate.kt [hunk 1]
```diff
@@ -26,18 +26,29 @@ class JooqPaginate(query: SelectWindowStep<*>, private var count: Int, private v
...
         this.orderField = orderField
     }
 
+    fun count(countQuery: SelectWindowStep<*>) {
+        countNum =  countQuery.fetchOneInto(Long::class.java) ?: -1L
+    }
     fun <T> find(clazz: Class<T>): JooqPage {
-        this.totalRow = this.query.count().toLong()
+        this.totalRow = if (countNum == -1L) {
+            this.query.count().toLong()
```
Comment (T-baby) [inline plugins/cloudopt-next-jooq/src/main/kotlin/net/cloudopt/next/jooq/JooqPaginate.kt:48]: It is preferable to change only this line of code.

File: plugins/cloudopt-next-jooq/src/main/kotlin/net/cloudopt/next/jooq/JooqPaginate.kt [hunk 2]
```diff
@@ -26,18 +26,29 @@ class JooqPaginate(query: SelectWindowStep<*>, private var count: Int, private v
...
-    private lateinit var orderField: OrderField<*>
+    private var countNum: Long = -1
+    private lateinit var orderField: MutableList<OrderField<*>>
 
     init {
         this.query = query
     }
 
-    fun order(orderField: OrderField<*>) {
+    fun order(vararg orderField: OrderField<*>) {
```
Comment (T-baby) [inline plugins/cloudopt-next-jooq/src/main/kotlin/net/cloudopt/next/jooq/JooqPaginate.kt:36]: This variable's name should be "orderFieldList".

File: plugins/cloudopt-next-jooq/src/main/kotlin/net/cloudopt/next/jooq/JooqPaginate.kt [hunk 3]
```diff
@@ -36,19 +35,13 @@ class JooqPaginate(query: SelectWindowStep<*>, private var count: Int, private v
...
+        this.orderField = orderFieldList
     }
+
     fun <T> find(clazz: Class<T>): JooqPage {
-        this.totalRow = if (countNum == -1L) {
-            this.query.count().toLong()
-        } else {
-            countNum
-        }
+        this.totalRow = JooqManager.dsl.selectCount().from(this.query).fetchOneInto(Int::class.java)?.toLong() ?: 0L
```
Comment (T-baby) [inline plugins/cloudopt-next-jooq/src/main/kotlin/net/cloudopt/next/jooq/JooqPaginate.kt:44]: Why do you want an integer variable before converting it to a long variable?

**Pull request merged**